htdocs/ticket/card.php

```diff
@@ -679,19 +679,27 @@
 	if ($action == "change_property" && GETPOST('btn_update_ticket_prop', 'alpha') && $permissiontoadd) {
 		$object->fetch(GETPOST('id', 'int'), '', GETPOST('track_id', 'alpha'));
 
-		$object->type_code = GETPOST('update_value_type', 'aZ09');
-		$object->severity_code = GETPOST('update_value_severity', 'aZ09');
-		$object->category_code = GETPOST('update_value_category', 'aZ09');
+		if (!GETPOST("update_value_type", 'alpha')) {
```

Comment:
> Sorry but I have changed my point of view and I revert my review. There still exist a problem when you edit the type field, you can reset the value if you select "-- Select type --" value. However this field must be mandatory.

This is due to the fact that these fields are not checked before updating.

The right way to fix this would be to use a function Ticket::validate() in create() and update(). But this function does not exist in DLB18 (I did not look after).

So I implemented a control in line with the code for ticket creation.